6.recursiveSum.js

```diff
@@ -1,17 +1,17 @@
 // Q6.
-// Implement a function that accepts an array of integers as an argument. 
-// This function should return the sum of each integer in the array. 
-// Your solution MUST be **recursive**. Your function must be pure (cannot use global variables)
-// 
+// Implement a function that accepts an array of integers as an argument.
+// This function should return the sum of each integer in the array.
+// Y`our solution MUST be **recursive**. Your function must be pure (cannot use global variables)
+//
 // Example:
 // const arr1 = [5, 2, 9, 11]
 // recursiveSum(arr1) should return 27
 
-
 const recursiveSum = (arr) => {
+  if (arr.length <= 1) return arr.pop();
+  return arr.pop() + recursiveSum(arr);
+};
```

Comment:
This problem asked for a function to be pure. Pop mutates it. So if we call the function twice with the same array we wont get a consistent answer.